java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -4207,4 +4207,38 @@ public void blmpop_timeout_check(BaseClient client) {
                             testClient.blmpop(new String[] {key}, PopDirection.LEFT, 0).get(3, TimeUnit.SECONDS));
         }
     }
+
+    @SneakyThrows
+    @ParameterizedTest(autoCloseArguments = false)
+    @MethodSource("getClients")
+    public void lset(BaseClient client) {
+        // setup
+        String key = "testKey";
+        String nonExistingKey = "nonExisting";
+        long index = 0;
+        long oobIndex = 10;
+        String element = "zero";
+        String[] lpushArgs = {"four", "three", "two", "one"};
+        String[] expectedList = {"zero", "two", "three", "four"};
+
+        // key does not exist
+        ExecutionException noSuchKeyException =
+                assertThrows(
+                        ExecutionException.class, () -> client.lset(nonExistingKey, index, element).get());
+        assertInstanceOf(RequestException.class, noSuchKeyException.getCause());
+
+        // pushing elements to list
+        client.lpush(key, lpushArgs).get();
+
+        // index out of range
+        ExecutionException indexOutOfBoundException =
```

Comment:
Add a test with negative index. Does -1 causes an error or replaces the last element?

Reply:
last element